README.md

```diff
@@ -1,6 +1,53 @@
 # patch-filters
 special project to host GSOC project Patch Suppressions 
 
+## PatchFilter Description
+
+### SuppressionPatchXpathFilter
+
+Filter SuppressionPatchXpathFilter(TreeWalker level) has three different strategies that control suppression
+scope. if property `strategy` is set to `newline`, then it only accepts TreeWalker audit events for TreeWalker 
+Check violations whose line number belong to added lines in patch file and will suppress all TreeWalker Checks’ 
+violation messages which are not from added lines. if property `strategy` is set to `patchedline`, it will accept 
+all violations whose line number belong to added/changed lines in patch file. if property `strategy` is set to `context`, 
+for checks listed in `supportContextStrategyChecks`, it will not only accept violations whose line number belong to added/changed/deleted lines,
+but also consider a wider context that new code introduces violations outside of added/changed lines, but its child nodes in added/changed lines,
+for checks not listed in `supportContextStrategyChecks`, it will accept violations whose line number belong to added/changed/deleted lines in patch file.
+If there is no configured patch file or the optional is set to true and patch file was not found the Filter suppresses all audit events.
+
```

Comment:
Please make a note if it is possible to use all three properties (`checkNameForContextStrategyByTokenOrParentSet`, `supportContextStrategyChecks`, `neverSuppressedChecks`) and what's the effect.

Reply:
done